wafer/talks/views.py

```diff
@@ -231,13 +231,20 @@ def form_valid(self, form):
         version.object_repr = str(review)
         version.save()
 
+        # Update the talk to 'under consideration' if a review is
+        # added.
         talk = review.talk
         if talk.status == SUBMITTED:
             talk.status = UNDER_CONSIDERATION
             with revisions.create_revision():
                 revisions.set_user(self.request.user)
                 revisions.set_comment("Status changed by review process")
                 talk.save()
+            # We also change the date to apotentially changing the talk to
+            # ensure is_current behaves as expected
+            talk_version = Version.objects.get_for_object(talk).order_by('-pk').first()
+            version.revision.date_created = talk_version.revision.date_created
+            version.revision.save()
```

Comment:
Urgh, this is a little hacky. Can't we update the status before we save the review to avoid having to hack the talk?

Reply:
Yeah, that change works. More code, but less race-y.